src/components/Toolbar/index.jsx

```diff
@@ -0,0 +1,121 @@
+/* global cozy */
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Menu, { Item } from 'components/Menu'
+import styles from './index.styl'
+
+import { withBreakpoints } from 'cozy-ui/react'
+
+const { BarRight } = cozy.bar
+
+class Toolbar extends Component {
+  render() {
+    const {
+      children,
+      moreMenuTitle,
+      moreMenuDisabled,
+      moreMenuClassName,
+      moreMenuButton,
+      breakpoints: { isMobile }
+    } = this.props
+
+    const actions = React.Children.toArray(children)
+    const highlightedActions = actions.filter(child => child.props.highlighted)
+    const secondaryActions = actions.filter(child => !child.props.highlighted)
+
+    const MoreMenu = (
+      <Menu
+        title={moreMenuTitle}
+        disabled={moreMenuDisabled}
+        className={moreMenuClassName}
+        button={moreMenuButton}
+      >
+        {highlightedActions.map(component =>
+          this.wrapInItem(
+            React.cloneElement(component, {
+              highlighted: false,
+              isSecondaryAction: false
+            })
+          )
+        )}
+        {secondaryActions.map(component =>
+          this.wrapInItem(
+            React.cloneElement(component, {
+              highlighted: false,
+              isSecondaryAction: true
+            })
+          )
+        )}
+      </Menu>
+    )
+
+    return (
+      <div role="toolbar">
+        {highlightedActions.map(component =>
+          React.cloneElement(component, {
+            highlighted: true,
+            isSecondaryAction: false
+          })
+        )}
+        {isMobile ? <BarRight>{MoreMenu}</BarRight> : MoreMenu}
+      </div>
+    )
+  }
+
+  wrapInItem(component) {
+    return <Item>{component}</Item>
+  }
```

Comment:
Can't we merge this code and the `cloneElement` one. I mean:

```jsx
const wrap = comp => <div>{comp}</div>

const WrappedComp = wrap(React.cloneElement(<Item />, { highlighted: true }))
```

is the same as

```jsx
const WrappedComp = props => (
  <div>
    <Item highlighted={true} />
  </div>
)
```

but the second one is simpler, doesn't it?

Reply:
It would be simpler, but we need the `React.cloneElement`.

Reply:
That is my very question:

Do we ever need `React.cloneElement` within JSX if the purpose is not to keep `ref` value?